django/contrib/admin/sites.py

```diff
@@ -65,10 +65,14 @@ def __init__(self, name='admin'):
         self._global_actions = self._actions.copy()
         all_sites.add(self)
 
-    def check(self):
+    def check(self, app_configs):
+        if app_configs is not None:
+            app_configs = set(app_configs)  # Speed up lookups below
+
         errors = []
         for admin_obj in six.itervalues(self._registry):
-            errors.extend(admin_obj.check())
+            if app_configs is None or admin_obj.model._meta.app_config in app_configs:
```

Comment:
This is weird behaviour, why can't the checks framework just pass the list of all installed app configs? This would allow check functions to not have two branches 🤔